packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -60,7 +60,7 @@ class ProtocolDiscovery {
   void _handleLine(String line) {
     Uri uri;
 
-    final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on (http://[^ \n]+)');
+    final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on ((http:|[/][/]|www.)([a-z]|[A-Z]|[:0-9]|[/.])*)');
```

Comment:
It should never be `www`, and you don't need square brackets around the individual slashes, which would make the first part `(http:|//)`

In the second part, You can just put all the acceptable characters in one "or" group, like so: `[a-zA-Z0-9:/\.]`

Finally, you want to require something in the second part, so you'd use `+` and not `*`

That would make the whole thing:

```dart
final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on ((http|//)[a-zA-Z0-9:/\.]+)');
```

Reply:
Ah, we need to add the open and close brackets, and apparently equals to the list of legal characters:

```dart
[a-zA-Z0-9:/=\.\[\]]
```

Reply:
Dart was not working, I added a couple of escape more and it seems to work with this one:
`((http|\/\/)[a-zA-Z0-9:/=\.\\[\\]]+)`